lib/__tests__/fixtures/postprocess-update-text.mjs

```diff
@@ -0,0 +1,33 @@
+export default {
+	name: 'update-rule-text',
+
+	/** @type {(result: LintResult) => void} */
+	postprocess(result) {
+		const updatedWarnings = result.warnings.map((warning) => {
+			let updatedWarning = { ...warning };
+
+			// Find the rule that the warning is on and update the warning text to include it
+			result._postcssResult.root.walk((node) => {
```

Comment:
[suggest] I don't want to encourage using the internal API (`_postcssResult`). If it's useful, how about adding an argument for `PostCSS.Root`?

```js
postprocess(result, root) {
  // result is stylelint.LinterResult
  // root is postcss.Root
}
```

Reply:
Yes sounds good. This is the updated Processor type:
```js
        export type Processor = () => {
                name: string;
                postprocess: (result: LintResult, root?: PostCSS.Root) => void;
        };
```
https://github.com/YuanboXue-Amber/stylelint/blob/985bdf1a7f7cd7999867f97153ca4cc440dbb643/types/stylelint/index.d.ts#L206-L209
I updated this test accordingly.